juju/juju
# [JUJU-2627] Allow testing on a mac

The following allows testing without relying on dqlite. This means that instead of dqlite, we inject sqlite. For local development running `go test ./...` now works as it did previously. This cleans up the developer workflow and makes juju a lot more tenable for developing on.

By default all the makefile targets will use dqlite, so to opt out of that you have to override the build tags.

## Checklist


## QA steps

No need to go through the make file to trigger a test.

```
$ go test -v ./database/... -check.v 
```
